app/src/main/cpp/ExternalVR.cpp

```diff
@@ -128,17 +128,23 @@ struct ExternalVR::State {
   vrb::Vector eyeOffsets[device::EyeCount];
   uint64_t lastFrameId;
   bool firstPresentingFrame;
+  bool compositorEnabled;
+  bool waitingForExit;
 
   State() : deviceCapabilities(0) {
     memset(&data, 0, sizeof(mozilla::gfx::VRExternalShmem));
     memset(&system, 0, sizeof(mozilla::gfx::VRSystemState));
     memset(&browser, 0, sizeof(mozilla::gfx::VRBrowserState));
     data.version = mozilla::gfx::kVRExternalVersion;
     data.size = sizeof(mozilla::gfx::VRExternalShmem);
-    pthread_mutex_init(&(data.systemMutex), nullptr);
-    pthread_mutex_init(&(data.browserMutex), nullptr);
-    pthread_cond_init(&(data.systemCond), nullptr);
-    pthread_cond_init(&(data.browserCond), nullptr);
+    pthread_mutex_init(&data.systemMutex, nullptr);
+    pthread_mutexattr_t attr;
+    pthread_mutexattr_init(&attr);
+    pthread_mutexattr_settype(&attr, PTHREAD_MUTEX_RECURSIVE);
```

Comment:
I would prefer we avoid recursive locks (http://www.fieryrobot.com/blog/2008/10/14/recursive-locks-will-kill-you/)

Why is this needed?